src/lockscreen.js

```diff
@@ -72,34 +81,49 @@ var Lockscreen = GObject.registerClass({
     }
 
     set lock(lock) {
-        if ('_lock' in this && this._lock === lock)
+        if (this._lock === lock)
             return;
         if (this._lockChangedId !== 0)
             this._manager.disconnect(this._lockChangedId);
         this._lockChangedId = this._manager.connect(
             `changed::${lock}`, this._updateLockStateWithLock.bind(this));
         this._lock = lock;
+
+        var path = `/app/share/lockscreens/${lock}`;
```

Comment:
It would be great to respect the build prefix, or do some kind of relative lookup.